homeassistant/components/ituran/sensor.py

```diff
@@ -0,0 +1,119 @@
+"""Sensors for Ituran vehicles."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from datetime import datetime
+
+from pyituran import Vehicle
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+)
+from homeassistant.const import (
+    DEGREE,
+    UnitOfElectricPotential,
+    UnitOfLength,
+    UnitOfSpeed,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+
+from . import IturanConfigEntry
+from .coordinator import IturanDataUpdateCoordinator
+from .entity import IturanBaseEntity
+
+
+@dataclass(frozen=True, kw_only=True)
+class IturanSensorEntityDescription(SensorEntityDescription):
+    """Describes Ituran sensor entity."""
+
+    value_fn: Callable[[Vehicle], StateType | datetime]
+
+
+SENSOR_TYPES: list[IturanSensorEntityDescription] = [
+    IturanSensorEntityDescription(
+        key="address",
+        translation_key="address",
+        entity_registry_enabled_default=False,
+        value_fn=lambda vehicle: vehicle.address,
+    ),
+    IturanSensorEntityDescription(
+        key="battery_voltage",
+        translation_key="battery_voltage",
+        device_class=SensorDeviceClass.VOLTAGE,
+        native_unit_of_measurement=UnitOfElectricPotential.VOLT,
+        suggested_display_precision=0,
+        entity_registry_enabled_default=False,
+        value_fn=lambda vehicle: vehicle.battery_voltage,
+    ),
+    IturanSensorEntityDescription(
+        key="heading",
+        translation_key="heading",
+        native_unit_of_measurement=DEGREE,
+        suggested_display_precision=0,
+        entity_registry_enabled_default=False,
+        value_fn=lambda vehicle: vehicle.heading,
+    ),
+    IturanSensorEntityDescription(
```

Comment:
I don't like that sensor tbh, each entity is tracking a `last_update` anway, so we can just get rid of this

Reply:
Those aren't exactly the same. The entity's `last_update` tells you when the value was last changed whereas this sensor will tells you when said value was last valid/true, which may be closer in time.
If, for example, one of the sensors was last updated 5 hours ago. How would you know if it's because its value really didn't change or because the car didn't publish any information?

BTW - This information is also different from the time at which the web service was polled.

In any case, it's disabled by default so it shouldn't interfere with anyone who isn't interested with this extra information.

Reply:
so what exactly is it? The last time ituran sent data to the web for you to grab? In that case I'd think of a better name, like last update from car or smething like that

Reply:
> so what exactly is it? The last time ituran sent data to the web for you to grab

Yes, exactly.

> In that case I'd think of a better name, like last update from car or smething like that

OK, changed to `Last update from vehicle`